### PR TITLE
Performance optimalization

### PR DESCRIPTION
**Pull request by Timowest12**
Project by me @Timowest12  and @phelian23 
- Used EXPLAIN ANALYZE queries to check performance

- created indexes to speed up performance

-Took screen-shots for the execution time before and after

### CODE REVIEW

**Comment by Timowest12** [comment]
Before optimalisation:
SELECT COUNT(*) FROM visits where animal_id = 4;


Querie: CREATE INDEX animals_id_index ON visits(animals_id);


Solution:

**Comment by Timowest12** [comment]
Before optimalization:
SELECT * FROM visits where vet_id = 2;


CREATE INDEX email_index ON owners(email);

solution

**Comment by Timowest12** [comment]
Before optimalization:
SELECT * FROM owners where email = 'owner_18327@mail.com';


querie: CREATE INDEX vets_id_index ON visits(vets_id);


solution:

File: schema.sql [hunk 1]
```diff
@@ -54,4 +54,14 @@ CREATE TABLE visits (
...
    CONSTRAINT fk_vets FOREIGN KEY(vets_id) REFERENCES vets(id)
-);
\ No newline at end of file
+);
+
+
+INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;
+
+
+insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.com';
```
Comment (addod19) [inline schema.sql]: - [ ] It would be better to move the `insert` query operation into the `data.sql` file

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @Timowest12 and @phelian23, 

Good job so far! Queries implemented with their respective optimization code :100: Screenshots provided to aid presentation :1st_place_medal: 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Writing more descriptive commit messages to describe our block of code

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
### Hi @phelian23   🙋‍♀️, 

# STATUS APPROVED ✔
   
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**